### ISSUE TITLE
Remove deprecated config flags

### ISSUE THREAD

**Issue by alexec** [argoproj/argo-workflows#7971]
# Summary

Several values have been deprecated since 2019 and can be removed straight away.

* executorImage
* executorImagePullPolicy
* executorResources
* namespace

Looking at the code, they're probably broken anyway.

Changes will need to be made:

* https://github.com/argoproj/argo-workflows/blob/980f2feb7b887b23513f1fc0717321bfdf134506/config/config.go#L29
* https://github.com/argoproj/argo-workflows/blob/master/docs/upgrading.md - meniton the breaking change

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

**Comment by AnuragThePathak** [comment]
Hello! I would like to work on this issue for GSoC 2022. Can you please show me around or tell me a little how should I proceed with it?

**Comment by terrytangyuan** [comment]
@AnuragThePathak Please read this page first: https://github.com/argoproj/argo-workflows/blob/master/docs/mentoring.md

**Comment by alexec** [comment]
I think this too small issue for GSoC, but fine for general mentoring.

**Comment by alexec** [comment]
I've added notes for where to make the changes.

**Comment by AnuragThePathak** [comment]
In order to fix the dependency issues created after removing the above-mentioned fields in the Config struct of config.go file, I created a function that returns the executor with Config as the receiver. I took the help of your PR in the process.
But I was having a problem with [this part(https://github.com/argoproj/argo-workflows/blob/200ae9c032dd6d6510d5cd9a1317cd1f9d60bc53/workflow/controller/workflowpod.go#L702). Should I remove it as you did?
